MaxText/llama4_ckpt_unscanned.py

```diff
@@ -0,0 +1,562 @@
+# Copyright 2025 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      https://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Convert weights from a Llama4 PyTorch model to a MaxText one.
+
+Usage:
+
+Get LLaMA chkpt_vars from Meta
+
+Example cmd:
+To save a ckpt
+JAX_PLATFORMS=CPU python -m  MaxText.llama4_ckpt_unscanned --base-model-path [CHKPT_DIR] \
+ --maxtext-model-path [OUTPUT_CHKPT_DIR] --model-size llama4-17b-16e
+
+The base model checkpoints should be in the format `{name}.{chkpt_idx}.pth`
+For example: `llama4-17b-16e.00.pth`
+This script requires a large memory VM (most likely a CPU instance with at least 500GB of memory).
+
+NOTE @jacobplatin: eventually, we'll want to merge this into the regular `llama_or_mistral` script
+since the logic is effectively the same.
+
+Also note that we'll want to add HF chkpt support eventually as well.
+"""
+# pylint: disable=g-line-too-long
+import argparse
+import gc
+import logging
+import os
+import pathlib
+import re
+from dataclasses import dataclass
+
+os.environ["JAX_PLATFORMS"] = "cpu"
+
+import ml_dtypes
+import numpy as np
+import psutil
+import torch
+from tqdm import tqdm
+
+from MaxText import max_logging
+from MaxText.inference_utils import str2bool
+from MaxText.llama_or_mistral_ckpt import save_weights_to_checkpoint, permute_to_match_maxtext_rope
+
+MODEL_PARAMS_DICT = {
```

Comment:
It may be error-prone when having 2 configs, one here and one in llama_or_mixtral_ckpt. Could we reuse the one in  llama_or_mixtral and have some assertion for llama4 supported model type?